universal_binaries break building when replace is true

### What happened?

Hello!

I believe that having `universal_binaries/replace: true ` while using `--output` flag in build command results in an ugly out of index error.

```
./bin/goreleaser build --clean --snapshot --single-target --output .
  • starting build...
  • loading config file                              file=.goreleaser.yml
  • building only for darwin/arm64                   reason=single target is enabled
  • loading environment variables
  • getting and validating git state
    • building...                                    commit=ac16a6b74f895188d0a01361fb80cdaad88645ad latest tag=v1.0.0
    • pipe skipped                                   reason=disabled during snapshot mode
  • parsing tag
  • setting defaults
      • DEPRECATED: `archives.rlcp` will be the default soon, check https://goreleaser.com/deprecations#archivesrlcp for more info
      • DEPRECATED: `nfpms.replacements` should not be used anymore, check https://goreleaser.com/deprecations#nfpmsreplacements for more info
  • running before hooks
    • running                                        hook=./packaging/completions.sh
    • took: 3s
  • snapshotting
    • building snapshot...                           version=1.0.0-SNAPSHOT-ac16a6b
  • checking distribution directory
    • cleaning dist
  • loading go mod information
  • build prerequisites
  • writing effective config file
    • writing                                        config=dist/config.yaml
  • building binaries
    • building                                       binary=dist/forwarder_darwin_arm64/forwarder
    • took: 1s
  • universal binaries
    • creating from 1 binaries                       binary=dist/forwarder_darwin_all/forwarder id=forwarder
  • storing release metadata
    • writing                                        file=dist/artifacts.json
    • writing                                        file=dist/metadata.json
  • copying binary to "."
panic: runtime error: index out of range [0] with length 0

goroutine 41 [running]:
github.com/goreleaser/goreleaser/cmd.withOutputPipe.Run({{0x16ef4785b?, 0x1011305a8?}}, 0x1400013dd80?)
        /Users/hubert/go/pkg/mod/github.com/goreleaser/goreleaser@v1.17.2/cmd/build.go:217 +0xcc
github.com/goreleaser/goreleaser/internal/middleware/errhandler.Handle.func1(0x103ed7a00?)
        /Users/hubert/go/pkg/mod/github.com/goreleaser/goreleaser@v1.17.2/internal/middleware/errhandler/error.go:15 +0x24
github.com/goreleaser/goreleaser/internal/middleware/logging.Log.func1(0x102ce9ac0?)
        /Users/hubert/go/pkg/mod/github.com/goreleaser/goreleaser@v1.17.2/internal/middleware/logging/logging.go:28 +0x128
github.com/goreleaser/goreleaser/cmd.buildProject.func1()
        /Users/hubert/go/pkg/mod/github.com/goreleaser/goreleaser@v1.17.2/cmd/build.go:113 +0x214
github.com/caarlos0/ctrlc.(*Ctrlc).Run.func1()
        /Users/hubert/go/pkg/mod/github.com/caarlos0/ctrlc@v1.2.0/ctrlc.go:48 +0x30
created by github.com/caarlos0/ctrlc.(*Ctrlc).Run
        /Users/hubert/go/pkg/mod/github.com/caarlos0/ctrlc@v1.2.0/ctrlc.go:47 +0x84
```

When I disable the replace option and run build again it succeeds . 

### How can we reproduce this?

Please visit https://github.com/saucelabs/forwarder. 

The goreleaser config is here: https://github.com/saucelabs/forwarder/blob/main/.goreleaser.yml

Steps to reproduce: 
1. `git clone git@github.com:saucelabs/forwarder.git`
2. `cd forwarder`
3. `make install-dependencies` or manually install goreleaser in the right version 
4. `make build` or `./bin/goreleaser build --clean --snapshot --single-target --output .`
5. See the error

### goreleaser version

```bash
goreleaser version dev
goos: darwin
goarch: arm64
module version: v1.17.2, checksum: h1:31zA06Sz6/zMNDMvI7l6I4XwFMRoI/WnTMiX8Puu6tE=
```


### GoReleaser Check

- [X] goreleaser check shows no errors

### Search

- [X] I did search for other open and closed issues before opening this.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Additional context

_No response_

--- a/cmd/build.go
+++ b/cmd/build.go
@@ -178,6 +178,7 @@ func setupBuildSingleTarget(ctx *context.Context) {
 		build.Goamd64 = nil
 		build.Targets = nil
 	}
+	ctx.Config.UniversalBinaries = nil
 }
 
 func setupBuildID(ctx *context.Context, ids []string) error {
@@ -214,7 +215,11 @@ func (w withOutputPipe) String() string {
 }
 
 func (w withOutputPipe) Run(ctx *context.Context) error {
-	path := ctx.Artifacts.Filter(artifact.ByType(artifact.Binary)).List()[0].Path
+	bins := ctx.Artifacts.Filter(artifact.ByType(artifact.Binary)).List()
+	if len(bins) == 0 {
+		return fmt.Errorf("no binary found")
+	}
+	path := bins[0].Path
 	out := w.output
 	if out == "." {
 		out = filepath.Base(path)

--- a/cmd/build_test.go
+++ b/cmd/build_test.go
@@ -292,15 +292,19 @@ func TestBuildSingleTargetWithSpecificTargets(t *testing.T) {
 				},
 			},
 		},
+		UniversalBinaries: []config.UniversalBinary{
+			{Replace: true},
+		},
 	})
 
-	t.Setenv("GOOS", "linux")
+	t.Setenv("GOOS", "darwin")
 	t.Setenv("GOARCH", "amd64")
 	setupBuildSingleTarget(ctx)
 	require.Equal(t, config.Build{
-		Goos:   []string{"linux"},
+		Goos:   []string{"darwin"},
 		Goarch: []string{"amd64"},
 	}, ctx.Config.Builds[0])
+	require.Nil(t, ctx.Config.UniversalBinaries)
 }
 
 func TestBuildSingleTargetRemoveOtherOptions(t *testing.T) {

